health/health.go

```diff
@@ -159,33 +179,39 @@ func CheckStatus() map[string]string { // TODO(stevvooe) this needs a proper typ
 }
 
 // Register associates the checker with the provided name. We allow
-// overwrites to a specific check status.
-func Register(name string, check Checker) {
-	mutex.Lock()
-	defer mutex.Unlock()
-	_, ok := registeredChecks[name]
+// overwrites to a specific check status. registry may be nil to use the
+// default registry.
+func Register(registry *Registry, name string, check Checker) {
```

Comment:
Users should still call `health.Register`, which is the same as `health.DefaultRegistry.Register` without the first argument.